finagle-netty4/src/main/scala/com/twitter/finagle/netty4/Netty4Listener.scala

```diff
@@ -128,11 +128,16 @@ private[finagle] case class Netty4Listener[In, Out](
 
         // The boss loop immediately starts refusing new work.
         // Existing tasks have ``deadline`` time to finish executing.
-        bossLoop
+        val shutdown: NettyFuture[_] = bossLoop
           .shutdownGracefully(0 /* quietPeriod */ , deadline.inMillis /* timeout */ , TimeUnit.MILLISECONDS)
-          .addListener(new GenericFutureListener[Nothing] {
-            def operationComplete(future: Nothing): Unit = p.setDone()
-          })
+        def shutdownListener[N <: NettyFuture[_]]: GenericFutureListener[N] =
```

Comment:
ffti: `NettyFuture[Any]` and `GenericFutureListener[Any]` should work as well.

I have an internal patch that also fixes that but it's been soaking for review for a long time so I'm totally fine to merge this in.